### PR TITLE
Fix 417

### LINKED ISSUE #417: [BUG] ph5api.cut should exit with error or skip trace if data assigned to none type is encountered
**Describe the bug**
ph5api.cut exits with the message ` ERROR: object of type 'NoneType' has no len()` it if called on data segment that contains gaps. This error message is confusing and should be replaced with a concise message that explains why the program is exiting or the gap should be skipped and the next gap should be cut. 

**Environment (please complete the following information):**
 - ph5api 2019.93
- Centos 7

**To Reproduce**
On expirment 14-032 
call ph5toms -p ./ -n master.ph5 --station MG06 --channel HHN --starttime 2014-07-19T00:00:00 --stoptime 2014-07-22T23:59:59

**Expected behavior**
Bets case, gap should be skipped and PH5 cut should continue to operate. Next concise error exception should be thrown. 

Traceback (most recent call last):
  File "/home/PIC/.conda/envs/ph5/bin/ph5toms", line 11, in <module>
    load_entry_point('ph5', 'console_scripts', 'ph5toms')()
  File "/home/PIC/ph5_environment/PH5/ph5/clients/ph5toms.py", line 1224, in main
    for stream in ph5ms.process_all():
  File "/home/PIC/ph5_environment/PH5/ph5/clients/ph5toms.py", line 1012, in process_all
    stream = self.create_trace(cut)
  File "/home/PIC/ph5_environment/PH5/ph5/clients/ph5toms.py", line 549, in create_trace
    apply_time_correction=nt, das_t=das)
  File "/home/PIC/ph5_environment/PH5/ph5/core/ph5api.py", line 1282, in cut
    calc_stop_fepoch = trace_start_fepoch + (len(data) / sr)
TypeError: object of type 'NoneType' has no len()
Closing remaining open files:./miniPH5_00096.ph5...done./master.ph5...done

### PR DESCRIPTION
**Pull request by timronan**
Thank your for contributing to PH5

Before submitting a PR, please review the pull request guidelines:
https://github.com/PIC-IRIS/PH5/blob/master/CONTRIBUTING.md#submitting-a-pull-request
Pull request to replace PR #423 one my commits did not make it into this issues. 

### What does this PR do?
Address issue 417 by skipping traces that do not contain data. 
fixes #417 
### Relevant Issues?
Issue 417
### Checklist
- [X ] This PR is not directly related to an existing issue (which has no PR yet).
- [ X] All tests pass. Matches test cases that fail in master reported in issue #422 
- [ X] Any new or changed features have are documented.
No Changes features, just less buggy. 
- [X ] Changes have been added to `CHANGELOG.txt` .
- [ X] First time contributors have added your name to `CONTRIBUTORS.txt` .

### CODE REVIEW

**Comment by timronan** [comment]
This is ready for review.

File: ph5/core/ph5api.py [hunk 1]
```diff
@@ -1283,7 +1283,11 @@ def cut(self, das, start_fepoch, stop_fepoch, chan=1,
                     das_t.append(d)
         # adjust the number of data samples as to not over extend the
         # cut_stop_fepoch
-        calc_stop_fepoch = trace_start_fepoch + (len(data) / sr)
+        if data is None:
+            return [Trace(np.array([]), start_fepoch, 0., 0, sample_rate,
+                          None, None, das_t, None, None, clock=clock)]
+        else:
```
Comment (dsentinel) [inline ph5/core/ph5api.py:1289]: I don't think this else block is necessary the above if returns.

File: ph5/core/ph5api.py [hunk 2]
```diff
@@ -1283,7 +1283,11 @@ def cut(self, das, start_fepoch, stop_fepoch, chan=1,
                     das_t.append(d)
         # adjust the number of data samples as to not over extend the
         # cut_stop_fepoch
-        calc_stop_fepoch = trace_start_fepoch + (len(data) / sr)
+        if data is None:
```
Comment (dsentinel) [inline ph5/core/ph5api.py:1286]: You're duplicating https://github.com/PIC-IRIS/PH5/blob/9a7d7657e28e5f880fd2be12a32bb0300401e89f/ph5/core/ph5api.py#L1300
Should it be just moved up here, and deleted below?

Comment (timronan) [reply ph5/core/ph5api.py:1286]: The second `if data is None: ` was the original check and was left to make minimal changes. It seems okay to just move it and delete.

Comment (timronan) [reply ph5/core/ph5api.py:1286]: The duplicated block and the else statement have been removed by commit 0e453b2d464ec0ee66d50df36919d1d70872c98e. Let me know if I should change anything else.

**Comment by dsentinel** [comment]
@timronan Would it make sense to add a test?

**Comment by timronan** [comment]
Yeah we should probably add a test to this. I will work on it soon.

**Comment by timronan** [comment]
Are there any additional steps that should be taken fro this PR to be accepted?

**Comment by damhuonglan** [comment]
> Are there any additional steps that should be taken fro this PR to be accepted?

Look good to me. I already approved it. Just wait for @dsentinel .

**Comment by damhuonglan** [comment]
@timronan According to our software meeting today, you can merge this PR with only one approval. So it's ok for you to merge this now.

File: ph5/core/ph5api.py (hunk 1 above)
Comment (timronan) [reply ph5/core/ph5api.py]: This seems to be resolved. Is there anything else that I should do so the requested changes can be closed?

**Comment by timronan** [comment]
Hi @damhuonglan, 

I have a few questions to make sure that I am doing the pull request process correctly. Can you verify that I am supposed to merge my pull request. Am I supposed to do a squash and merge? I can't find anything in the 
https://github.com/PIC-IRIS/PH5/blob/master/CONTRIBUTING.md#submitting-a-pull-request page that outlines how I am supposed to merge my branch. 


Currently my ability to merge is blocked because I have outstanding 'Changes Requested.' I think that I resolved all of these requested changes but will continue to make code changes if necessary. Please let me know how to proceed.

**Review by damhuonglan — APPROVED** [review]

**Comment by damhuonglan** [comment]
@timronan I already marked all change requests as resolved for you. However, recently the travis-ci check got stuck for any PRs that have new commits that prevent merging. At first I thought that is for a specific PR, but now it happens to yours too. I will talk with Lloyd about this. He has been too busy recently and may be slow on this. Please be patient.

**Comment by timronan** [comment]
@damhuonglan I think you are correct and that Travis CI is stuck for this PR. No problem on the timing, I know Lloyd is busy, let me know if there is anything I need to do.

**Comment by damhuonglan** [comment]
@timronan Travis CI is not stuck anymore. But for Lloyd's requested change, it doesn't go away after I marked at resolved. Please look at the part that he request to change, it isn't marked as Outdated, you should change something in that part and commit to make it Outdated. That will remove the requested change left that prevent the merging.

**Review by dsentinel — APPROVED** [review]

**Comment by timronan** [comment]
@damhuonglan and/or @dsentinel is this pull request ready to merge? Which merging pattern should be used to merge a PH5 pull request?

**Pull request merged**